### PR TITLE
feat: support import attributes

### LINKED ISSUE #20294: type of dynamic url imports nested in a dynamic import always resolves to `any`
# Summary

type of dynamic imports from URL like `import("https://<url>").Member` are always `any`. 

## Reproduction

image

```ts
// @deno-types="https://esm.sh/v131/@octokit/rest@20.0.1/dist-types/index.d.ts"
import { Octokit } from "https://esm.sh/@octokit/rest@20.0.1"
import type { AssertFalse, IsAny } from "https://deno.land/std@0.200.0/testing/types.ts"

type _ = AssertFalse<IsAny<Octokit>>
```
(adding `@deno-types` makes no differences)

```sh
 ~/r/d/test  deno check main.ts     2023년 08월 27일 (일) 오전 12시 25분 43초
Check file:///home/scarf/repo/deno-packages/test/main.ts
error: TS2344 [ERROR]: Type 'true' does not satisfy the constraint 'false'.
type _ = AssertFalse<IsAny<Octokit>>
                     ~~~~~~~~~~~~~~
    at file:///home/scarf/repo/deno-packages/test/main.ts:5:22
```
this does not seem to be a esm.sh problem as manually converting dynamic imports to import statement works:
```ts
// typechecks
import { Constructor } from "https://esm.sh/v131/@octokit/core@5.0.0/dist-types/types.d.ts"
import { PaginateInterface } from "https://esm.sh/v131/@octokit/plugin-paginate-rest@8.0.0/dist-types/index.d.ts"
import { RestEndpointMethods } from "https://esm.sh/v131/@octokit/plugin-rest-endpoint-methods@9.0.0/dist-types/generated/method-types.d.ts"
import { Api } from "https://esm.sh/v131/@octokit/plugin-rest-endpoint-methods@9.0.0/dist-types/types.d.ts"

import { Octokit as Core } from "https://esm.sh/v131/@octokit/core@5.0.0/dist-types/index.d.ts"
export type { RestEndpointMethodTypes } from "https://esm.sh/v131/@octokit/plugin-rest-endpoint-methods@9.0.0/dist-types/index.d.ts"
export declare const Octokit:
	& typeof Core
	& Constructor<{ paginate: PaginateInterface } & RestEndpointMethods & Api>
export type Octokit = InstanceType<typeof Octokit>

import type { AssertFalse, IsAny } from "https://deno.land/std@0.200.0/testing/types.ts"

type _ = AssertFalse<IsAny<Octokit>>
```

`npm:` works:

image

```ts
// typechecks
import { Octokit } from "npm:@octokit/rest@20.0.1"
import type { AssertFalse, IsAny } from "https://deno.land/std@0.200.0/testing/types.ts"

type _ = AssertFalse<IsAny<Octokit>>
```

### LINKED ISSUE #20345: Enable JSX Namespace support
If you try to use a namespaced tag or attribute in a .tsx file you hit the following error:

> TypeError: JSX Namespace is disabled by default because react does not support it yet. You can specify jsc.transform.react.throwIfNamespace to false to override default behavior at ...

For reference, the namespace syntax looks like this:

```jsx
<foo:bah/>
<div foo:bah="..."/>
```

This seems to be a config in swc... https://swc.rs/docs/configuration/compilation#jsctransformreactthrowifnamespace

I can't see anywhere to override this in Deno, could this config be exposed or ideally just default it to false, as usage of JSX shouldn't be limited to whatever React does or doesn't support.

It also seems to be inconsistent with TypeScript checking, which doesn't throw up any issues when checking a module containing this syntax.

### PR DESCRIPTION
**Pull request by dsherret**
Upgrades to deno_ast 0.29.

No idea if these tests will pass because I can't run the test suite locally.

Closes #20294
Closes #20345

### CODE REVIEW

File: cli/lsp/diagnostics.rs [hunk 1]
```diff
@@ -1084,8 +1084,8 @@ impl DenoDiagnostic {
     let (severity, message, data) = match self {
       Self::DenoWarn(message) => (lsp::DiagnosticSeverity::WARNING, message.to_string(), None),
       Self::ImportMapRemap { from, to } => (lsp::DiagnosticSeverity::HINT, format!("The import specifier can be remapped to \"{to}\" which will resolve it via the active import map."), Some(json!({ "from": from, "to": to }))),
-      Self::InvalidAssertType(assert_type) => (lsp::DiagnosticSeverity::ERROR, format!("The module is a JSON module and expected an assertion type of \"json\". Instead got \"{assert_type}\"."), None),
-      Self::NoAssertType => (lsp::DiagnosticSeverity::ERROR, "The module is a JSON module and not being imported with an import assertion. Consider adding `assert { type: \"json\" }` to the import statement.".to_string(), None),
+      Self::InvalidAttributeType(assert_type) => (lsp::DiagnosticSeverity::ERROR, format!("The module is a JSON module and expected an assertion type of \"json\". Instead got \"{assert_type}\"."), None),
```
Comment (bartlomieju) [inline cli/lsp/diagnostics.rs:1087]: The error is `InvalidAttributeType` while the error message says `and expected an assertion type`; should we unify them?

File: cli/tsc/00_typescript.js [hunk 2]
```diff
@@ -33956,7 +33957,11 @@ ${lanes.join("\n")}
         function parseAssertClause(skipAssertKeyword) {
           const pos = getNodePos();
           if (!skipAssertKeyword) {
-            parseExpected(132 /* AssertKeyword */);
+	    if (token() === 118 /* WithKeyword */) {
+              parseExpected(118 /* AssertKeyword */);
```
Comment (bartlomieju) [inline cli/tsc/00_typescript.js:33961]:
```suggestion
              parseExpected(118 /* WithKeyword */);
```

File: tools/lint.js [hunk 3]
```diff
@@ -19,7 +19,8 @@ if (Deno.args.includes("--rs")) {
 if (!didLint) {
   await Promise.all([
     dlint(),
-    dlintPreferPrimordials(),
+    // todo(dsherret): re-enable
+    // dlintPreferPrimordials(),
```
Comment (dsherret) [inline tools/lint.js:23]: FYI @petamoriken -- I temporarily disabled this because there were about 50 lint errors.

**Review by bartlomieju — APPROVED** [review]
LGTM, are the primordial lints really bad?

File: tools/lint.js (hunk 3 above)
Comment (petamoriken) [reply tools/lint.js]: OK, I'll check it.

**Pull request merged**